base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -305,9 +315,54 @@ Result<void> FetchHostPackage(BuildApi& build_api, const Build& build,
   std::string host_tools_filepath =
       CF_EXPECT(build_api.DownloadFile(build, target_dir, host_tools_name));
   trace.CompletePhase("Download", FileSize(host_tools_filepath));
+
   CF_EXPECT(
       ExtractArchiveContents(host_tools_filepath, target_dir, keep_archives));
   trace.CompletePhase("Extract");
+
+  std::unique_ptr<Runfiles> runfiles;
+  if (!debian_package_executables.empty()) {
+    std::string self_path;
+    CF_EXPECT(android::base::Readlink("/proc/self/exe", &self_path));
+    std::string runfiles_error;
+    runfiles.reset(
+        Runfiles::Create(self_path, BAZEL_CURRENT_REPOSITORY, &runfiles_error));
+    CF_EXPECTF(runfiles.get(), "Runfiles::Create failed: '{}'", runfiles_error);
+  }
+  for (const std::string& executable : debian_package_executables) {
+    std::size_t last_slash = executable.rfind("/");
+    std::string_view short_name = executable;
+    if (last_slash != std::string::npos) {
+      short_name = short_name.substr(last_slash + 1);
+    }
+    CF_EXPECT(!short_name.empty());
+    if (short_name[0] == ':') {
+      short_name = short_name.substr(1);
+    }
+
+    std::string from_runfiles = runfiles->Rlocation(executable);
+    std::string from_install =
+        fmt::format("/usr/lib/cuttlefish-common/bin/{}", short_name);
+
+    std::string source;
+    if (!from_runfiles.empty() && FileExists(from_runfiles)) {
+      source = from_runfiles;
```

Comment:
What happens if I run global `cvd` from my bazel repository? Not `bazel run cvd` but just plain `cvd` to invoke `/usr/bin/cvd`.
Will it use the binaries from that bazel repository or the ones from /usr/bin?

Reply:
Good point, I'm honestly not sure what it would do in that situation.

I've changed it to pull from `/usr/lib/cuttlefish-common/bin` when the executable itself is `/usr/lib/cuttlefish-common/bin/cvd`, and otherwise to try the `Runfiles` strategy.

Reply:
I suppose that's fine, though a bit hacky (what happens if we change the location where cvd is installed on the host? or if hard link it instead of symlink?).
A better approach would be to have a build-time flag to disable the runfiles feature and pass this flag when building cvd as part of the debian package.

Reply:
My other concern is what happens if I execute the `cvd` built by bazel outside the bazel repository and without `bazel run`?
I often do this to test the `cvd create` flow from a local android build.

Reply:
> My other concern is what happens if I execute the `cvd` built by bazel outside the bazel repository and without `bazel run`?

If you don't `cp` or `mv` it somewhere, so that it is run it out of `bazel-bin/cuttlefish/host/commands/cvd/cvd`, it behaves comparably to `bazel run`.